docs(jbod): kraft: expands description for managing metadata storage (#10083)
strimzi/strimzi-kafka-operator: documentation/modules/configuring/ref-storage-jbod.adoc

== Configuring the storage volume used to store the KRaft metadata log

> In KRaft mode, a Kafka cluster stores its metadata in a log present on all nodes, including brokers and controllers.

Reviewer: I wonder if this would be a bit better wording as not every metadata log is equal and on the brokers it is AFAIU only a local copy and not the main metadata log as with the controllers.

```suggestion
In KRaft mode, a copy of the metadata log is stored on every node, including brokers and controllers.
```

> Each node reserves one of its data volumes for the KRaft metadata log. 

Reviewer: `reserves` suggests to me that it is used only for the metadata which is wrong. Maybe this would sound better?

```suggestion
Each node uses one of its data volumes for the KRaft metadata log. 
```

By default, the log is stored on the volume with the lowest ID. 
However, you can specify another volume using the `kraftMetadata` property.

> For controller-only nodes, which don't handle data, dedicating a single volume to the metadata log is generally sufficient for storage. 

Reviewer: I would make it more clear that this means JBOD makes little sense here.

```suggestion
For controller-only nodes, which don't handle data, the storage will be used only for the metadata log.
The metadata log will be always stored only on one volume, so using JBOD storage with multiple volumes does not improve the performance or increase the effectively available disk space. 
```

> Meanwhile, for broker nodes or nodes combining broker and controller roles, although they may require multiple volumes, they still share the same volume for storing both the metadata log and partition replica data. 

Reviewer: ```suggestion
Meanwhile, broker nodes or nodes combining broker and controller roles, will share the same volume for storing both the metadata log and partition replica data.
They can also utilize JBOD storage with multiple volumes as one of the volumes will be shared by the metadata log and partition replica data and any additional volumes will be used for partition replica data only.
```
